Plugins/NativeCapture/Source/NativeCapture.cpp

```diff
@@ -4,15 +4,123 @@
 
 #include <Babylon/JsRuntime.h>
 #include <GraphicsImpl.h>
+#include <FrameBuffer.h>
 
 #include <vector>
 
+namespace
+{
+    using FrameCallback = std::function<void(uint32_t width, uint32_t height, bgfx::TextureFormat::Enum format, bool yFlip, gsl::span<const uint8_t> data)>;
+    using FrameProviderCleanup = std::function<void()>;
+    using FrameProviderTicket = gsl::final_action<FrameProviderCleanup>;
+
+    FrameProviderTicket BeginFrameCapture(Babylon::Graphics::Impl& graphicsImpl, bgfx::FrameBufferHandle frameBufferHandle, FrameCallback callback)
+    {
+        class DefaultBufferFrameProvider final : public std::enable_shared_from_this<DefaultBufferFrameProvider>
+        {
+        public:
+            static FrameProviderTicket Create(Babylon::Graphics::Impl& graphicsImpl, FrameCallback callback)
+            {
+                std::shared_ptr<DefaultBufferFrameProvider> frameProvider{new DefaultBufferFrameProvider(graphicsImpl, std::move(callback))};
```

Comment:
does make_shared not work?

Reply:
It works if I make the constructor public, but that doesn't express the intent well, and we instantiate these types infrequently, so the extra cost is not worth the extra confusion in my opinion.

Reply:
okay